cmd/antrea-agent/agent.go

```diff
@@ -323,31 +324,48 @@ func run(o *Options) error {
 		)
 	}
 
-	var mcRouteController *mcroute.MCRouteController
+	var mcDefaultRouteController *mcroute.MCDefaultRouteController
 	var mcStrechedNetworkPolicyController *mcroute.StretchedNetworkPolicyController
+	var mcPodRouteController *mcroute.MCPodRouteController
 	var mcInformerFactory mcinformers.SharedInformerFactory
+	var mcInformerFactoryWithOption mcinformers.SharedInformerFactory
+
 	if enableMulticlusterGW {
-		mcNamespace := env.GetPodNamespace()
-		if o.config.Multicluster.Namespace != "" {
-			mcNamespace = o.config.Multicluster.Namespace
-		}
-		mcInformerFactory = mcinformers.NewSharedInformerFactory(mcClient, informerDefaultResync)
-		gwInformer := mcInformerFactory.Multicluster().V1alpha1().Gateways()
-		ciImportInformer := mcInformerFactory.Multicluster().V1alpha1().ClusterInfoImports()
-		mcRouteController = mcroute.NewMCRouteController(
+		mcInformerFactoryWithOption = mcinformers.NewSharedInformerFactoryWithOptions(mcClient,
+			informerDefaultResync,
+			mcinformers.WithTweakListOptions(func(lo *metav1.ListOptions) {
+				lo.FieldSelector = fields.Set{"metadata.namespace": o.config.Multicluster.Namespace}.String()
+			}),
+		)
+		mcDefaultRouteController = mcroute.NewMCDefaultRouteController(
 			mcClient,
-			gwInformer,
-			ciImportInformer,
+			mcInformerFactoryWithOption,
 			ofClient,
 			ovsBridgeClient,
 			ifaceStore,
 			nodeConfig,
-			mcNamespace,
+			o.config.Multicluster.Namespace,
 			o.config.Multicluster.EnableStretchedNetworkPolicy,
 			o.config.Multicluster.EnablePodToPodConnectivity,
 		)
+		if networkConfig.TrafficEncapMode != config.TrafficEncapModeEncap {
+			if !networkConfig.IPv4Enabled && networkConfig.IPv6Enabled {
```

Comment:
This check should be moved to right after `if enableMulticlusterGW {`.

Reply:
`&& networkConfig.IPv6Enabled` can be removed?

Reply:
Fixed.